### PR TITLE
remove `unalias` TableScan filters when create Physical Filter

### PR DESCRIPTION
**Pull request by jackwener**
## Which issue does this PR close?

Closes #.

## Rationale for this change

- remove `unalias` TableScan filters
- refactor  CreateExternalTable
- fix typo

### CODE REVIEW

File: datafusion/core/src/physical_planner.rs [hunk 1]
```diff
@@ -562,8 +562,7 @@ impl DefaultPhysicalPlanner {
                     // doesn't know (nor should care) how the relation was
                     // referred to in the query
                     let filters = unnormalize_cols(filters.iter().cloned());
-                    let unaliased: Vec<Expr> = filters.into_iter().map(unalias).collect();
```
Comment (jackwener) [inline datafusion/core/src/physical_planner.rs:565]: It's useless

Comment (alamb) [reply datafusion/core/src/physical_planner.rs:565]: I think we have to rely on our test coverage here, and given all the existing tests pass I agree!

**Review by alamb — APPROVED** [review]
Thank you @jackwener  -- this is a very nice PR 👏

File: datafusion/core/src/physical_planner.rs (hunk 1 above)
Comment (jackwener) [reply datafusion/core/src/physical_planner.rs:565]: Yes, theoretically and in terms of code, the alias in the filter expression does not affect the schema or output.

**Pull request merged**